server/runtime/src/main/java/org/infinispan/server/Bootstrap.java

```diff
@@ -99,44 +105,88 @@ protected void handleArgumentCommand(String command, String parameter, Iterator<
+      try {
+
+         if (!serverRoot.isAbsolute()) {
+            serverRoot = serverRoot.getAbsoluteFile();
+         }
+         File confDir = new File(serverRoot, Server.DEFAULT_SERVER_CONFIG);
+         if (configurationFile == null) {
+            configurationFile = new File(confDir, Server.DEFAULT_CONFIGURATION_FILE);
+         } else if (!configurationFile.isAbsolute()) {
+            configurationFile = Paths.get(confDir.getPath(), configurationFile.getPath()).toFile();
+         }
+         File logDir = new File(serverRoot, Server.DEFAULT_SERVER_LOG);
+         properties.putIfAbsent(Server.INFINISPAN_SERVER_LOG_PATH, logDir.getAbsolutePath());
+
+         if (loggingFile == null) {
+            loggingFile = new File(confDir, Server.DEFAULT_LOGGING_FILE);
+         } else if (!loggingFile.isAbsolute()) {
+            loggingFile = Paths.get(confDir.getPath(), loggingFile.getPath()).toFile();
+         }
 
-      logJVMInformation();
+         System.setProperty("log4j.configurationFile", loggingFile.getAbsolutePath());
+         if (!loggingFile.canRead()) {
+            throw new IllegalStateException(String.format("Cannot read %s", loggingFile));
+         }
 
-      Runtime.getRuntime().addShutdownHook(new ShutdownHook(exitHandler));
-      Server.log.serverStarting(Version.getBrandName());
-      Server.log.serverConfiguration(configurationFile.getAbsolutePath());
-      Server.log.loggingConfiguration(loggingFile.getAbsolutePath());
-      try (Server server = new Server(serverRoot, configurationFile, properties)) {
+         logger = getBootLogger();
+
+         logJVMInformation(logger);
+
+         Runtime.getRuntime().addShutdownHook(new ShutdownHook(exitHandler));
+         Server.log.serverStarting(Version.getBrandName());
+         Server.log.serverConfiguration(configurationFile.getAbsolutePath());
+         Server.log.loggingConfiguration(loggingFile.getAbsolutePath());
+
+         server = new Server(serverRoot, configurationFile, properties);
          server.setExitHandler(exitHandler);
          server.run().get();
-      } catch (Exception e) {
-         Server.log.serverFailedToStart(Version.getBrandName(), e);
+
+      } catch (Exception e) { // when starting, all failures should be handled
+
+         if (logger != null) {
+            Server.log.serverFailedToStart(Version.getBrandName(), e);
+         } else {
+            // @Message(value = "%s Server failed to start", id = 80028)
+            // ISPN080028: brandName Server failed to start
+            stdErr.printf("ISPN%5d: %s Server failed to start -> %s", 80028, Version.getBrandName(), e.getMessage());
+         }
+
+      } finally {
+
+         if (server != null)
+            server.close();
       }
    }
 
+   private Logger getBootLogger() {
+
+      List<String> errors = new ArrayList<>(0);
+
+      // the listener call will be in the same getLogger thread
+      StatusConsoleListener listener = new StatusConsoleListener(Level.ERROR) {
```

Comment:
instead of using the listener, you can inspect the logger output:
`StatusLogger.getLogger().getStatusData()
        .stream()
        .filter(s -> s.getLevel().isMoreSpecificThan(Level.ERROR))
        .forEach(// log to stderr  );`